### DISCUSSION TITLE
Getting a secret in python is either wrong implemented or the documentation is wrong.

### DISCUSSION THREAD

**Discussion by pebneter** [pulumi/pulumi#9914] [General]
### What happened?

I want to write some custom function in my pulumi __main.py__.
In the function, I need access to a secret stored in the pulumi config.

According to documentation (https://www.pulumi.com/docs/intro/concepts/secrets/), I should do this:
```
import pulumi

config = pulumi.Config()

print(config.require('name'))
print(config.require_secret('dbPassword'))
```
But when doing it according to documentation, my function only gets an output object, not the actual value `<pulumi.output.Output object at 0x0000027361CF5210`.
In the Pulumi slack, I found this post: https://pulumi-community.slack.com/archives/CDE799L1M/p1655476974371809 which pointed me to the solution.
Instead of reading the secret with `config.require_secret('dbPassword')`, I need to read it with `config.require('dbPassword')`.

So either the documentation is incorrect, or the implementation is.

### Steps to reproduce

Follow the documentation at https://www.pulumi.com/docs/intro/concepts/secrets/ and set a configuration value with `pulumi config set --secret dbPassword S3cr37`

Try to access it with `config.require_secret('dbPassword')`

### Expected Behavior

Returns secret in plain text

### Actual Behavior

Returns output object.

### Versions used

Python 3.10.5
pulumi==3.34.1
pulumi-auth0==2.9.0

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by AaronFriel** [comment]
Hey @pebneter, this is expected behavior due to our implementation of [secrets as outputs](https://www.pulumi.com/docs/intro/concepts/secrets/#how-secrets-relate-to-outputs). Outputs are kind of like promises or futures, and carry some metadata like "is this value secret". We use that secret-ness to ensure that your secret values - passwords, API keys, and so on - are not stored unencrypted in our service or in your state file.

You can pass these secrets as inputs to resources and the Pulumi engine will take care of "unwrapping" them for the resource. 

If you _must_ obtain the value as plain text, for example to print it out as output, you can use [apply](https://www.pulumi.com/docs/intro/concepts/inputs-outputs/#apply) to obtain the inner value and pass it to a function.

```python
def print_plain_value(value):
  print("this function receives a plain, non-'Output' value, like so:")
  print(value)

config.require_secret('dbPassword').apply(lambda x: print_plain_value(x))
# Could also be:                   .apply(print_plain_value)
```

The Pulumi engine however depends on keeping secrets and outputs "wrapped" for as long as possible, not just for encryption, so I would caution you to avoid creating resources inside a function like `print_plain_value`.

**Reply by pebneter** [reply]
Thanks for the example - I will try it in my scenario. I need a secret value to authenticate against a REST API and tried to do it with `require_secret` but I couldn't make it work.
Ist this a case where I need the plaintext secret or would it be possible to avoid it?

I was just confused that my code ran fine when using just `require` even when it is stored as secret. I'm not sure it that is supposed to work, because I could nowhere find information about it.

**Reply by pebneter** [reply]
I tried to use your approach, but on my method to call the API for authentication, I only get `'client_secret': <pulumi.output.Output object at 0x0000010B671C0>` instead of the password and the call fails.
